plotly_resampler/aggregation/aggregation_interface.py

```diff
@@ -4,169 +4,260 @@
-                )
-        return s
-
-    def _replace_gap_end_none(self, s: pd.Series) -> pd.Series:
-        # ------- REPLACE None where a gap ends -------
-        med_diff, s_idx_diff = self._calc_med_diff(s)
-        if med_diff is not None:
-            # Replace data-points with None where the gaps occur
-            # The default is the end of a gap
-            nan_mask = s_idx_diff > 4 * med_diff
-            if self.nan_position == "begin":
-                # Replace the last non-gap datapoint (begin of gap) with Nan
-                nan_mask = np.roll(nan_mask, -1)
-            elif self.nan_position == "both":
-                # Replace the encompassing gap datapoints with Nan
-                nan_mask |= np.roll(nan_mask, -1)
-            s.loc[nan_mask] = None
-        return s
-
-    def aggregate(self, s: pd.Series, n_out: int) -> pd.Series:
-        """Aggregate (downsample) the given input series to the given n_out samples.
+        Parameters
+        ----------
+        x : np.ndarray, optional
+            The x-axis data, by default None
+        y : np.ndarray
+            The y-axis data
+        n_out : int, optional
+            The number of output datapoints, by default None
+        kwargs : dict
+            Additional keyword arguments that are passed to the `_aggregate` method.
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The aggregated x and y data, respectively
+
+        """
+        assert n_out is not None
+
+        # assert isinstance(y, np.ndarray)
+        assert y.ndim == 1
+        DataAggregator._supports_dtype(y, self.y_dtype_regex_list)
+
+        if x is not None:
+            assert x.ndim == 1
+            assert x.shape == y.shape
+            DataAggregator._supports_dtype(x, self.x_dtype_regex_list)
+
+        return self._aggregate(x, y, n_out=n_out, **kwargs)
+
+
+class DataPointSelector(AbstractAggregator, ABC):
+    """Implementation of the AbstractAggregator interface for data point selection.
+
+    A data point selector is an aggregator that selects data points, and thus doesn't
+    aggregate the data.
+
+    This class utilizes the `arg_downsample` method to compute the index positions.
+    """
+
+    @abstractmethod
+    def _arg_downsample(
+        self,
+        x: Optional[np.ndarray] = None,
+        y: Optional[np.ndarray] = None,
+        n_out: int = None,
+        **kwargs,
+    ) -> np.ndarray:
+        raise NotImplementedError
+
+    def arg_downsample(
+        self,
+        x: Optional[np.ndarray] = None,
+        y: np.ndarray = None,
+        n_out: int = None,
+        **kwargs,
+    ) -> np.ndarray:
+        """Compute the index positions for the downsampled representation.
 
         Parameters
         ----------
-        s: pd.Series
-            The series that has to be aggregated.
+        x: Optional[np.ndarray]
+            The time dimension of the to-be-aggregated series
+        y: Optional[np.ndarray]
```

Comment:
Let's not specify the types here because we've got them in the signature. Also this type is mismatched.